packages/amplify-cli/src/index.ts

```diff
@@ -48,11 +50,15 @@ export async function run() {
         throw new Error(verificationResult.message);
       }
     }
-
     rewireDeprecatedCommands(input);
     const context = constructContext(pluginPlatform, input);
+    await attachTelemetry(context);
+    errorHandler = boundErrorHandler.bind(context);
```

Comment:
Why do we need `bind` to context instead of calling `context.telemetry.emitXXXX methods? what is the advantage?

Reply:
I need the context to be present when it fails on SIGINT